hilti/runtime/src/context.cc

```diff
@@ -13,8 +13,14 @@ using namespace hilti::rt::detail;
 
 namespace hilti::rt::context::detail {
 
-// Not part of global state, it's per thread.
+// Not part of global state, it's per thread. (See comment in `logging.h` about
+// platform differences for TLS.)
+#if defined(__linux__)
+__thread Context* __current = nullptr;
```

Comment:
If we had some `#define THREAD_LOCAL ...` in some header (also a nice place to put documentation) we could avoid some of the checks added in this patch.

Reply:
Ack, I guess that's what `util.h` is for.